### ISSUE TITLE
bug(python): 

### ISSUE THREAD

**Issue by Phat-Loc** [lancedb/lancedb#771]
### LanceDB version

0.4.3

### What happened?

Program hangs when using cuda as accelerator to build an index.

table.create_index(metric="dot", num_partitions=1024, accelerator='cuda')

Control never comes back from `tensor = torch.tensor(np_tensor)` in 

https://github.com/lancedb/lance/blob/3ae29b8bd298f0ca5808efb719b5ecb34e829ecb/python/python/lance/torch/data.py#L52

I tried this on two different machines. I am using CUDA Version: 12.3. I suspect it has to do with Cuda and multiprocessing. I recall seeing a separate process being spawned but it wasn't using the pytorch multiprocessing.

https://pytorch.org/docs/stable/notes/multiprocessing.html

Please let me know if you experience the same issues. It could be my setup.

### Are there known steps to reproduce?

Run 

table.create_index(metric="dot", num_partitions=1024, accelerator='cuda')

**Comment by eddyxu** [comment]
This was due to a recent change that requires you to run the following code to set multiprocessing for cuda (https://pytorch.org/docs/stable/notes/multiprocessing.html#cuda-in-multiprocessing)

```
from multiprocessing import set_start_method

if __name__ == "__main__":
    set_start_method("spawn")
```



We are re-evaluating this approach, we might either revert the change or give options to enable async loader (but by default it is turned off).

**Comment by Phat-Loc** [comment]
`if __name__ == '__main__':
    set_start_method("spawn", force=True)`

works should I close the issue?

**Comment by eddyxu** [comment]
I will work on this tomorrow revert the change. Let's keep this one open?

**Comment by eddyxu** [comment]
Fixed in https://github.com/lancedb/lance/commit/4423096881f3bee804de84ccd8564de1fc546d23
